### ISSUE TITLE
Chat user search doesn't work sometimes

### ISSUE THREAD

**Issue by ForNeVeR** [telegramdesktop/tdesktop#7417]
### Steps to reproduce
1. Open a supergroup chat
2. Try to search for member (type `@` and begin of member nickname)
3. For some members in some chats, it won't work

For me, it doesn't work for a bot member (`@<redacted>_bot`) in supergroups with >700 and >4000 members. Other people report that they cannot find me (alongside the bot) in the same chats.

### Expected behaviour
User name should still be suggested no matter the group size and other circumstances. It works in mobile clients, but doesn't work in tdesktop

### Actual behaviour
Nothing is suggested.

### Configuration
**Operating system:** Windows 10.

**Version of Telegram Desktop:** latest stable, 1.9.21.

**Used theme**: default, Night Mode is enabled.

**Comment by jeffythedragonslayer** [comment]
Can you see any other nicknames in the member popup above the "Write a message..." field at all?

**Comment by ForNeVeR** [comment]
Yes, I can see some of them, it's just this bot I can't find.

The bot is named `@dotnet_<smth>_bot`, and there's a member named `@dotnet<smthelse>`.

When I type `@dot`, I can see `@dotnet<smthelse>` user, but not `@dotnet_<smth>_bot`. Even if I type almost fully `@dotnet_<smth>_`, then I see no completion.

The same bot user could be found in some other chats, but in this particular chat it couldn't be found. Interestingly enough, I can't find this particular user, `@dotnet<smthelse>`, in another chat.

**Comment by Aokromes** [comment]
i don't know if this maybe server bug.

**Comment by ForNeVeR** [comment]
I doubt it's a server bug. Mobile clients (which, presumably, use the same API) work well for the same cases when tdesktop doesn't. It's not only me who experiences this issue, there're other people reporting the same.

**Comment by Azizjan-leo** [comment]
Same issue

**Comment by ForNeVeR** [comment]
Still reproduces for me in random chats, only in tdesktop.

**Comment by ForNeVeR** [comment]
Still reproduces for me in random chats, only in tdesktop.

**Comment by ForNeVeR** [comment]
Still reproduces for me in random chats, only in tdesktop. And is pretty annoying.

**Comment by ILoveSkilledCoders** [comment]
Hello, this is a serious problem.

I need to see all of the tags to me and my own comments and I can NEVER find them.

In one group I find 3 out of how many?

In another group, none.

Now I've lost important information because I can't find anyone who tagged me or my own comments.

This is on desktop and I tried to search in mobile b/c of what the OP said above, but it doesn't work either. It finds the last comment, that's it. How do I get to all the other comments above my last comment?

And what about all the people who wrote to me?

Please fix this.

**Comment by ILoveSkilledCoders** [comment]
> Still reproduces for me in random chats, only in tdesktop. And is pretty annoying.

Obviously they don't plan to fix it - sigh

**Comment by RememberTheAir** [comment]
@ILoveSkilledCoders you can search for `@`, the search will return all the messages where you have been mentioned (replies + mentions by name/username). Not sure this is what you're looking for

**Comment by RememberTheAir** [comment]
Oh, by the way, since nobody mentioned this yet - _I think_ that the issue of users non popping out when you type the "@" to mention someone can be somehow fixed by opening the members list and scrolling a bit, to force the app to sync the members list. This is just a workaround of course

**Comment by ILoveSkilledCoders** [comment]
> @ILoveSkilledCoders you can search for `@` or `#m`, the search will return all the messages where you have been mentioned (replies + mentions by name/username). Not sure this is what you're looking for

Yes, this is what I said above & no, it's not doing that for me. I've never used # though or with quotes, just @ without quotes.

Are you saying I need to use single quotes?

**Comment by ILoveSkilledCoders** [comment]
> Oh, by the way, since nobody mentioned this yet - _I think_ that the issue of users non popping out when you type the "@" to mention someone can be somehow fixed by opening the members list and scrolling a bit, to force the app to sync the members list. This is just a workaround of course
Nope, didn't work.

In fact, now there's less entries than there were yesterday LOL Only 2 & the most recent 2, not the 3 that were there yesterday &  I posted a TON yesterday, like at least 8 times.

**Comment by ForNeVeR** [comment]
Still reproduces, the problem is not resolved.

**Comment by ILoveSkilledCoders** [comment]
Yup, search still isn't working properly.

**Comment by ForNeVeR** [comment]
Just checked, it is still problematic.

**Comment by ILoveSkilledCoders** [comment]
I haven't tried in a while, but the other day I was searching for something by keywords (not a name) & I couldn't find it. This was in a group with just me & one other person.

**Comment by ForNeVeR** [comment]
Still reproduces for me using the same steps, no changes.

**Comment by ForNeVeR** [comment]
The likely reason is in this code snippet: https://github.com/telegramdesktop/tdesktop/blob/77c2e12ebc8cbbeed878056ccf660425b83e944b/Telegram/SourceFiles/chat_helpers/field_autocomplete.cpp#L472-L474

I have several hypotheses about this.
1. The number of users in the chats I participate in may be high (up to 7000 users), and one `GetFullChat` request may just not return all the users. Perhaps we should send some other request to get all the users, or even send a request to get a filtered list of users, if possible?
2. There may be some internal limit on the amount of chat member data stored, and that request may easily overflow the limit. This would also explain why random users get lost from other views, such as the excluded members list.
3. It's also possible that the check is wrong: currently, it will only send the data request if there's no chat participant info collected. But what if we have _some_ data collected already, from some other means (such as users entering and leaving the chat)? In such case, a full data request will almost never be sent, and tdesktop will try to reuse the existing (not nearly full) data lists to seek for the user names to complete.

**Comment by voed** [comment]
Still reproduces even in group with ~50 members

**Comment by ForNeVeR** [comment]
You wouldn't believe, but yeah, this still reproduces for me.

**Comment by ForNeVeR** [comment]
The issue is still present.

**Comment by ForNeVeR** [comment]
Verified and the issue is still present. I will not surrender.

**Comment by ForNeVeR** [comment]
**I do not want to criticize people for not fixing issues.** Believe me, as an OSS maintainer myself, I personally have _significantly_ more issues I would like to fix than I have resources to do so. It is totally understandable that minor, non-breaking issues do not get prioritized. Moreover, I do believe that if somebody figures out this problem and brings a PR to fix it, the project maintainers would adopt such a PR (sadly, myself I didn't yet find time to work on this).

I would like to note, though, that I quite a lot dislike this bot's approach in particular.

It is ok to close issues if _you asked the users for some necessary information_, such as logs, steps to reproduce, etc., and _didn't get it in some reasonable time_.

It is not ok to close the issue if _you yourself_ don't have time to fix it. It should not time-out on the absense of project developer's reaction; only on the one from the user! **It is not correct to close a valid issue**: _it is still an issue_, even if there's no movement happening in it.

But this is not the only project that employs this misuse of the GitHub's stale workflow. It is possible to set up differently, but folks here decided they want it this way. Fine, I'll play by their rules as long as I have time.

**Comment by Aokromes** [comment]
> **I do not want to criticize people for not fixing issues.** Believe me, as an OSS maintainer myself, I personally have _significantly_ more issues I would like to fix than I have resources to do so. It is totally understandable that minor, non-breaking issues do not get prioritized. Moreover, I do believe that if somebody figures out this problem and brings a PR to fix it, the project maintainers would adopt such a PR (sadly, myself I didn't yet find time to work on this).
> 
> I would like to note, though, that I quite a lot dislike this bot's approach in particular.
> 
> It is ok to close issues if _you asked the users for some necessary information_, such as logs, steps to reproduce, etc., and _didn't get it in some reasonable time_.
> 
> It is not ok to close the issue if _you yourself_ don't have time to fix it. It should not time-out on the absense of project developer's reaction; only on the one from the user! **It is not correct to close a valid issue**: _it is still an issue_, even if there's no movement happening in it.
> 
> But this is not the only project that employs this misuse of the GitHub's stale workflow. It is possible to set up differently, but folks here decided they want it this way. Fine, I'll play by their rules as long as I have time.

it's not the 1st, 2nd, 5th, 10th, 25th time i have to close already fixed bugs because preston forgets to close them, that's why the bot exist, also it's only 1 time each 6 months

**Comment by ForNeVeR** [comment]
I am sorry, mr. bot, but I will never surrender until the issue is still active. Just tried — still reproduces.

**Comment by Aokromes** [comment]
i think it's reproducible on  https://t.me/TelegramDesktopTalk searching for preston.

**Comment by ForNeVeR** [comment]
Checked right now — the issue still reproduces for me in the latest stable version of tdesktop.
